rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeDebugProfileState.kt

```diff
@@ -0,0 +1,113 @@
+    }
+
+    override fun execute(executor: Executor, runner: ProgramRunner<*>, workerProcessHandler: DebuggerWorkerProcessHandler): ExecutionResult {
+        throw UnsupportedOperationException("Should use overload with session")
+    }
+
+    override fun execute(executor: Executor, runner: ProgramRunner<*>, workerProcessHandler: DebuggerWorkerProcessHandler, lifetime: Lifetime): ExecutionResult {
+        lifetime.onTermination {
+            if (!targetProcessHandler.isProcessTerminated)
+                targetProcessHandler.destroyProcess()
+        }
+
+        return DefaultExecutionResult(console, workerProcessHandler)
+    }
+
+    override fun createWorkerRunCmd(lifetime: Lifetime, helper: DebuggerHelperHost, port: Int): Promise<WorkerRunInfo> {
+        val result = lifetime.createNestedAsyncPromise<WorkerRunInfo>()
+
+        val runCommandLine = createEmptyConsoleCommandLine(exeConfiguration.parameters.useExternalConsole)
+            .withEnvironment(exeConfiguration.parameters.envs)
+            .withParentEnvironmentType(if (exeConfiguration.parameters.isPassParentEnvs) {
+                GeneralCommandLine.ParentEnvironmentType.CONSOLE
+            } else {
+                GeneralCommandLine.ParentEnvironmentType.NONE
+            })
+            .withExePath(exeConfiguration.parameters.exePath)
+            .withWorkDirectory(exeConfiguration.parameters.workingDirectory)
+            .withParameters(exeConfiguration.parameters.programParameters)
+
+        targetProcessHandler = if (exeConfiguration.parameters.useExternalConsole)
+            ExternalConsoleMediator.createProcessHandler(runCommandLine) as KillableProcessHandler
+        else
+            KillableProcessHandler(runCommandLine)
+        val commandLineString = runCommandLine.commandLineString
+        logger.info("Process started: $commandLineString")
+        targetProcessHandler.addProcessListener(object: ProcessAdapter() {
+            override fun processTerminated(event: ProcessEvent) = logger.info("Process terminated: $commandLineString")
+        })
+
+        console = createConsole(exeConfiguration.parameters.useExternalConsole, targetProcessHandler, commandLineString, executionEnvironment.project)
+        console.attachToProcess(targetProcessHandler)
+        targetProcessHandler.startNotify()
+
+        application.executeOnPooledThread {
+            UnityPlayerListener(project, {
+                if (!it.isEditor) {
+                    while (isAvailable(it.debuggerPort))
+                        Thread.sleep(1)
+
+                    UIUtil.invokeLaterIfNeeded {
+                        logger.trace("Connecting to Player with port: ${it.debuggerPort}")
+                        remoteConfiguration.port = it.debuggerPort
+                        result.setResult(createWorkerRunInfoFor(port, DebuggerWorkerPlatform.AnyCpu))
+                    }
+                }
+            }, {}, lifetime)
```

Comment:
What is the duration of this `Lifetime`? It controls how long you're polling for - does it get terminated as soon as the result is set?